gpioioctl/lineset.go

```diff
@@ -0,0 +1,406 @@
+			return line
+		}
+	}
+	return nil
+}
+
+// LineSetLine is a specific line in a lineset. Using a LineSetLine,
+// you can read/write to a single pin in the set using the PinIO
+// interface.
+type LineSetLine struct {
+	// The GPIO Line Number
+	number uint32
+	// The offset for this LineSet struct
+	offset    uint32
+	name      string
+	parent    *LineSet
+	direction LineDir
+	pull      gpio.Pull
+	edge      gpio.Edge
+}
+
+/*
+   gpio.Pin
+*/
+
+// Number returns the Line's GPIO Line Number. Implements gpio.Pin
+func (lsl *LineSetLine) Number() int {
+	return int(lsl.number)
+}
+
+// Name returns the line's name. Implements gpio.Pin
+func (lsl *LineSetLine) Name() string {
+	return lsl.name
+}
+
+func (lsl *LineSetLine) Function() string {
+	return "not implemented"
+}
+
+func (lsl *LineSetLine) Direction() LineDir {
+    return lsl.direction
+}
+
+func (lsl *LineSetLine) Edge() gpio.Edge {
+    return lsl.edge
+}
+
+/*
+gpio.PinOut
+*/
+// Out writes to this specific GPIO line.
+func (lsl *LineSetLine) Out(l gpio.Level) error {
+	var mask, bits uint64
+	mask = 1 << lsl.offset
+	if l {
+		bits |= mask
+	}
+	return lsl.parent.Out(bits, mask)
+}
+
+// PWM is not implemented because of kernel design.
+func (lsl *LineSetLine) PWM(gpio.Duty, physic.Frequency) error {
+	return errors.New("not implemented")
+}
+
+/*
+gpio.PinIn
+*/
+// Halt interrupts a pending WaitForEdge. You can't halt a read
+// for a single line in a LineSet, so this returns an error. Use
+// LineSet.Halt()
+func (lsl *LineSetLine) Halt() error {
+	return errors.New("you can't halt an individual line in a LineSet. you must halt the LineSet")
+}
+
+// In configures the line for input. Since individual lines in a
+// LineSet cannot be re-configured this always returns an error.
+func (lsl *LineSetLine) In(pull gpio.Pull, edge gpio.Edge) error {
+	return errors.New("a LineSet line cannot be re-configured")
+}
+
+// Read returns the value of this specific line.
+func (lsl *LineSetLine) Read() gpio.Level {
+	var mask uint64 = 1 << lsl.offset
+	bits, err := lsl.parent.Read(mask)
+	if err != nil {
+		log.Printf("LineSetLine.Read() Error reading line %d. Error: %s\n", lsl.number, err)
+		return false
+	}
+
+	return (bits & mask) == mask
+}
+
+// Return the line information in JSON format.
+func (lsl *LineSetLine) String() string {
+	return fmt.Sprintf("{\"Name\": \"%s\", \"Offset\": %d, \"Number\": %d, \"Direction\": \"%s\", \"Pull\": \"%s\", \"Edge\": \"%s\"}",
+		lsl.name,
+		lsl.offset,
+		lsl.number,
+		DirectionLabels[lsl.direction],
+		PullLabels[lsl.pull],
+		EdgeLabels[lsl.edge])
+}
+
+// WaitForEdge will always return false for a LineSetLine. You MUST
+// use LineSet.WaitForEdge()
+func (lsl *LineSetLine) WaitForEdge(timeout time.Duration) bool {
+	return false
+}
+
+// Pull returns the configured PullUp/PullDown value for this line.
+func (lsl *LineSetLine) Pull() gpio.Pull {
+	return lsl.pull
+}
+
+// DefaultPull - return gpio.PullNoChange. Reviewing the GPIO v2 Kernel
```

Comment:
Suggestion:

// DefaultPull return gpio.PullNoChange.
//
// The GPIO v2 Kernel doesn't support this.